cid.go

```diff
@@ -162,7 +187,8 @@ func (c *Cid) bytesV0() []byte {
 }
 
 func (c *Cid) bytesV1() []byte {
-	buf := make([]byte, 8+len(c.hash))
+	// two 8 bytes (max) numbers plus hash
+	buf := make([]byte, 2*binary.MaxVarintLen64+len(c.hash))
 	n := binary.PutUvarint(buf, c.version)
 	n += binary.PutUvarint(buf[n:], c.codec)
 	copy(buf[n:], c.hash)
```

Comment:
We should take and check the return value of this copy to be extra safe. Its fine to panic if its different than we expect, but we don't want to accidentally return less than the whole thing